src/ProjectService.Business/Commands/ProjectUsers/RemoveProjectUsersCommand.cs

```diff
@@ -1,44 +1,47 @@
 ﻿using System;
 using System.Collections.Generic;
 using System.Linq;
+using System.Net;
+using System.Threading.Tasks;
 using LT.DigitalOffice.Kernel.AccessValidatorEngine.Interfaces;
 using LT.DigitalOffice.Kernel.Constants;
 using LT.DigitalOffice.Kernel.Enums;
-using LT.DigitalOffice.Kernel.Exceptions.Models;
+using LT.DigitalOffice.Kernel.Helpers.Interfaces;
 using LT.DigitalOffice.Kernel.Responses;
 using LT.DigitalOffice.ProjectService.Business.Commands.ProjectUsers.Interfaces;
 using LT.DigitalOffice.ProjectService.Data.Interfaces;
 
 namespace LT.DigitalOffice.ProjectService.Business.Commands.ProjectUsers
 {
-  public class RemoveUsersFromProjectCommand : IRemoveUsersFromProjectCommand
+  public class RemoveProjectUsersCommand : IRemoveProjectUsersCommand
   {
     private readonly IUserRepository _repository;
     private readonly IAccessValidator _accessValidator;
+    private readonly IResponseCreater _responseCreater;
 
-    public RemoveUsersFromProjectCommand(
+    public RemoveProjectUsersCommand(
       IUserRepository repository,
-      IAccessValidator accessValidator)
+      IAccessValidator accessValidator,
+      IResponseCreater responseCreater)
     {
       _repository = repository;
       _accessValidator = accessValidator;
+      _responseCreater = responseCreater;
     }
 
-    // TODO make async
-
-    public OperationResultResponse<bool> Execute(Guid projectId, List<Guid> userIds)
+    public async Task<OperationResultResponse<bool>> ExecuteAsync(Guid projectId, List<Guid> userIds)
     {
-      if (userIds == null || !userIds.Any())
+      if (!(await _accessValidator.HasRightsAsync(Rights.AddEditRemoveProjects)))
```

Comment:
project admin can add  and remove users too

Reply:
fixed